### PR TITLE
Game state and associated logic

### PR DESCRIPTION
**Pull request by kjc9**
- Added a common.py module for common structs
- Added game_state.py which holds the game state and sim logic for a game
- Changed team_state.py to support the new game_state.py

### CODE REVIEW

File: src/common.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    CAUGHT_STEALINGS = 3
+
+    # batting stats
+    BATTER_STRIKEOUTS = 10
+    BATTER_HITS = 11
+    BATTER_SINGLES = 12
+    BATTER_DOUBLES = 13
+    BATTER_TRIPLES = 14
+    BATTER_HRS = 15
+    BATTER_PLATE_APPEARANCES = 16
```
Comment (tehstone) [inline src/common.py:18]: let's add At Bats as well

File: src/game_state.py [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+                # TODO: Add additional pre pitch batting events here as needed
+
+        # Required checks failed, no event triggers
+        return False
+
+    def resolve_o_no(self) -> bool:
+        if self.cur_batting_team.team_enum in team_pitch_event_map:
+            event, start_season, end_season, req_blood = team_pitch_event_map[self.cur_batting_team.team_enum]
+            if event == PitchEventTeamBuff.O_NO:
+                if self.strikes == 2 and \
```
Comment (tehstone) [inline src/game_state.py:431]: this doesn't seem to be accounting for the number of balls in the count

File: src/team_state.py [hunk 3]
```diff
@@ -77,9 +128,10 @@ def from_config(cls, team_state: Dict[str, Any]):
         outs_for_inning: int = team_state["outs_for_inning"]
         lineup: Dict[int, str] = team_state["lineup"]
         starting_pitcher: str = team_state["starting_pitcher"]
-        stats: Dict[str, Dict[str, float]] = team_state["stats"]
-        blood: Dict[str, str] = team_state["blood"]
-
+        stlats: Dict[str, Dict[FK, float]] = team_state["stlats"]
+        game_stats: Dict[str, Dict[Stats, float]] = team_state["game_stats"]
+        blood: Dict[str, int] = team_state["blood"]
+        cur_battter_pos: int = team_state["cur_batter_pos"]
```
Comment (tehstone) [inline src/team_state.py]: extra t in batter

File: src/team_state.py [hunk 4]
```diff
@@ -90,14 +142,41 @@ def from_config(cls, team_state: Dict[str, Any]):
...
         stats = self.stats[batter]
         return batter, stats
 
-    def get_player_stats_by_id(self, id: str) -> Dict[str, float]:
-        return self.stats[id]
+    def get_player_stats_by_id(self, player_id: str) -> Dict[str, float]:
+        return self.stats[player_id]
+
+    def next_batter(self) -> None:
+        if len(self.lineup) == self.cur_batter:
```
Comment (tehstone) [inline src/team_state.py]: this is comparing the length of the lineup against the id of the current batter, needs to either lookup the batter at the last lineup position or compare the length against cur_batter_pos

File: src/team_state.py [hunk 5]
```diff
@@ -90,14 +142,41 @@ def from_config(cls, team_state: Dict[str, Any]):
...
+            self.cur_batter_pos = 1
```
Comment (tehstone) [inline src/team_state.py:227]: I think this should be set to 0 here

File: src/team_state.py [hunk 6]
```diff
@@ -90,14 +142,41 @@ def from_config(cls, team_state: Dict[str, Any]):
...
+        else:
+            self.cur_batter_pos += 1
+        self.cur_batter = self.lineup[self.cur_batter_pos]
+
+    def update_stat(self, player_id: str, stat_id: Stats, value: float) -> None:
+        self.game_stats[player_id][stat_id] += value
+
+    def get_defense_feature_vector(self) -> List[float]:
+        return []
```
Comment (tehstone) [inline src/team_state.py]: I assume this is a todo still? same with the next 3

File: src/common.py (hunk 1 above)
Comment (kjc9) [reply src/common.py:18]: Done

File: src/game_state.py (hunk 2 above)
Comment (kjc9) [reply src/game_state.py:431]: Good catch.  Fixed

File: src/team_state.py (hunk 3 above)
Comment (kjc9) [reply src/team_state.py]: fixed.

File: src/team_state.py (hunk 4 above)
Comment (kjc9) [reply src/team_state.py]: Fixed

File: src/team_state.py (hunk 5 above)
Comment (kjc9) [reply src/team_state.py:227]: I'm inclined to leave the lineup position as 1-indexed.  The accesses to this dict aren't done off of anything that would by default be 0-indexed (ie positions in an array or list).

File: src/team_state.py (hunk 6 above)
Comment (kjc9) [reply src/team_state.py]: Yep.  Working on it.

**Pull request merged**